src/GeoCoding/OrbitState.jl

```diff
@@ -5,3 +5,102 @@ struct OrbitState
     velocity::Array{Float64,1}
 end 
 
+
+get_speed(state::OrbitState) = LinearAlgebra.norm(state.velocity)
+
+
+get_burst_mid_states(image::SingleLookComplex, interpolator) = [interpolator(t) 
+                                                            for t in get_burst_mid_times(image)]
+
+
+"""
+orbit_state_interpolator(orbit_states::Vector{OrbitState}, image::SARImage, 
+    polynomial_degree::Integer=4, margin::Integer = 3 )
+
+    Create a polynomial interpolation functions for orbit states valid in the time span
+    from image start time to image end time.
+
+    #Returns
+    Anonymous interpolation function. (Input: DateTime, Output: OrbitState)
+"""
+function orbit_state_interpolator(orbit_states::Vector{OrbitState}, image::SARImage, 
+    polynomial_degree::Integer=4, margin::Integer = 3 )
+    time_range = get_time_range(image)
+    
+    # check that the orbit states cover the image
+    @assert (orbit_states[1].time < time_range[1]) &&  (time_range[2] < orbit_states[end].time)
+    
+    orbit_state_interpolator(orbit_states, time_range; 
+    polynomial_degree=polynomial_degree, margin = margin )
+end
+
+function orbit_state_interpolator(orbit_states::Vector{OrbitState}, time_range::Tuple{DateTime,DateTime}; 
+    polynomial_degree::Integer=4, margin::Integer = 3 )
+
+
+    #select orbit states
+    selected_orbit_states = _select_orbit_states(orbit_states, time_range, margin)
+    @assert length(selected_orbit_states) > polynomial_degree
+
+    # Get times
+    selected_times = [element.time for element in selected_orbit_states];
+    time_0 = selected_times[1]
+    seconds = Float64.(Dates.value.(selected_times .- time_0))/1000
+
+    # normalise data
+    position = [ state.position[i] for i=1:3, state in selected_orbit_states ];
+    normalised_position, mean_position, std_position  = _normalise_dim2(position)
+    velocity = [ state.velocity[i] for i=1:3, state in selected_orbit_states ];
+    normalised_velocity, mean_velocity, std_velocity  = _normalise_dim2(velocity)
+
+    # fit polynomial
+    position_polynomial = [Polynomials.fit(seconds, normalised_position[i,:], polynomial_degree) for i =1:3]
+    velocity_polynomial = [Polynomials.fit(seconds, normalised_velocity[i,:], polynomial_degree) for i =1:3]
+
+    ## create interpolation function
+    interpolator = t -> 
+    begin
+        @assert (time_range[1] < t) && (t < time_range[2])
```

Comment:
I would add
`@assert condition "t is outside time_range"`
or something similar